### PR TITLE
Add RELEASE.md

### PR DESCRIPTION
**Pull request by consideRatio**
Mostly copied from https://github.com/jupyterhub/chartpress, but slightly adjusted relating to detailed version update instructions.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+1. Reset the `version_info` variable in
+   [_version.py](oauthenticator/_version.py) appropriately with a `dev` element
+   and make a commit with the message `back to dev`.
+
+1. Push your two commits to master along with the annotated tags referencing
+   commits on master.
+
+   ```
+   git push --follow-tags <upstream> master
+   ```
```
Comment (willingc) [inline RELEASE.md]: Let's add a conda-forge section too.

Feedstock: https://github.com/conda-forge/oauthenticator-feedstock

conda-forge should pick up a new pypi release and generate a PR against the feedstock. Someone should keep an eye out for this PR and its tests and then merge.

Comment (consideRatio) [reply RELEASE.md]: Thanks for giving an introduction to this! I've never worked with a conda-forge package!

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+   [choldgraf/github-activity](https://github.com/choldgraf/github-activity)
+   utility.
+
+1. Set the `version_info` variable in [_version.py](oauthenticator/_version.py)
+   appropriately and make a commit with message `release <tag>`.
+
+1. Create a git tag for the commit.
+
+   ```
+   git tag -a <tag> -m <tag> HEAD
```
Comment (minrk) [inline RELEASE.md]:
```suggestion
   git tag -a $TAG -m $TAG
```

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+   ```
+
+1. Package the release
+   ```
+   python3 setup.py bdist_wheel
```
Comment (minrk) [inline RELEASE.md]:
```suggestion
   python3 setup.py sdist bdist_wheel
```

File: RELEASE.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+   ```
+
+1. Upload it to PyPI
+   ```
+   twine upload dist/oauthenticator-<tag>-py3-none-any.whl
```
Comment (minrk) [inline RELEASE.md]:
```suggestion
   twine upload dist/*
```

File: RELEASE.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
+# How to make a release
+
+`oauthenticator` is a package [available on
+PyPI](https://pypi.org/project/oauthenticator/) and conda-forge. These are instructions on how
+to make a release on PyPI.
+
+For you to follow along according to these instructions, you need:
+- To be a maintainer of the [PyPI chartpress
```
Comment (minrk) [inline RELEASE.md]:
```suggestion
- To be a maintainer of the [PyPI oauthenticator
```

File: RELEASE.md (hunk 4 above)
Comment (consideRatio) [reply RELEASE.md]: I'm not sure if its important, but may may want to clean away old built distributions before if we upload all of them? Or add --skip-existing or a flag like that if we upload all?

WDYT?

File: RELEASE.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+1. Create a git tag for the commit.
+
+   ```
+   git tag -a $TAG -m $TAG
+   ```
+
+1. Package the release
+   ```
+   python3 setup.py sdist bdist_wheel
```
Comment (manics) [inline RELEASE.md]: I followed the setup from travis:
https://github.com/jupyterhub/oauthenticator/blob/0d2b44c2b7f08090ac5cb07c7577c02abdece480/.travis.yml#L11-L13

```$ python3 setup.py sdist bdist_wheel
/oauthenticator-venv/lib/python3.7/site-packages/setuptools/dist.py:475: UserWarning: Normalizing '0.10.0dev' to '0.10.0.dev0'
  normalized_version,
usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help

error: invalid command 'bdist_wheel'
```
Probably `pip install wheel` needs to be added to the instructions or a requirements file?

File: RELEASE.md (hunk 4 above)
Comment (consideRatio) [reply RELEASE.md]: Ah, by having a git clean step along with a .gitignore file containing dist/ we are good!

File: RELEASE.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+   commits on master.
+
+   ```shell
+   git push --follow-tags $ORIGIN master
+   ```
+
+1. Following the release to PyPI, an automated PR should arrive to
+   [conda-forge/oauthenticator-feedstock](https://github.com/conda-forge/oauthenticator-feedstock),
+   check for the tests to succeed on this PR and then merge it to successfully
+   update the package for `conda` on the conda-forge channel.
```
Comment (consideRatio) [inline RELEASE.md]: @willingc is this understanding correct?

**Comment by betatim** [comment]
Time to merge?

**Comment by consideRatio** [comment]
@betatim sure! I got a thumbs up from Carol about my understanding about conda-forge also!

**Pull request merged**